### DISCUSSION TITLE
Gradle can't build the first sample

### DISCUSSION THREAD

**Discussion by Navidda** [eclipse-edc/Samples#189] [Q&A] [answered]
Hello

I am an absolute newbie here. I tried to run the first sample and I ran into an error. Not sure how to work around it. I did not have java in my machine so I installed java 17 at first and got a similar error so I thought I should switch to java 11 and this is what I get after I installed java 11.

    ➜ Samples git:(main) ./gradlew clean basic:basic-01-basic-connector:build 

    FAILURE: Build failed with an exception.

    * What went wrong:
    A problem occurred configuring root project 'samples'.
    > Could not resolve all files for configuration ':classpath'.
    > Could not resolve org.eclipse.edc:edc-build:0.4.1.
        Required by:
            project : > org.eclipse.edc.edc-build:org.eclipse.edc.edc-build.gradle.plugin:0.4.1
        > No matching variant of org.eclipse.edc:edc-build:0.4.1 was found. The consumer was configured to find a library for use during runtime, compatible with Java 11, packaged as a jar, and its dependencies declared externally, as well as attribute 'org.gradle.plugin.api-version' with value '8.0' but:
            - Variant 'apiElements' capability org.eclipse.edc:edc-build:0.4.1 declares a library, packaged as a jar, and its dependencies declared externally:
                - Incompatible because this component declares a component for use during compile-time, compatible with Java 17 and the consumer needed a component for use during runtime, compatible with Java 11
                - Other compatible attribute:
                    - Doesn't say anything about org.gradle.plugin.api-version (required '8.0')
            - Variant 'javadocElements' capability org.eclipse.edc:edc-build:0.4.1 declares a component for use during runtime, and its dependencies declared externally:
                - Incompatible because this component declares documentation and the consumer needed a library
                - Other compatible attributes:
                    - Doesn't say anything about its target Java version (required compatibility with Java 11)
                    - Doesn't say anything about its elements (required them packaged as a jar)
                    - Doesn't say anything about org.gradle.plugin.api-version (required '8.0')
            - Variant 'runtimeElements' capability org.eclipse.edc:edc-build:0.4.1 declares a library for use during runtime, packaged as a jar, and its dependencies declared externally:
                - Incompatible because this component declares a component, compatible with Java 17 and the consumer needed
    a component, compatible with Java 11
                - Other compatible attribute:
                    - Doesn't say anything about org.gradle.plugin.api-version (required '8.0')
            - Variant 'sourcesElements' capability org.eclipse.edc:edc-build:0.4.1 declares a component for use during runtime, and its dependencies declared externally:
                - Incompatible because this component declares documentation and the consumer needed a library
                - Other compatible attributes:
                    - Doesn't say anything about its target Java version (required compatibility with Java 11)
                    - Doesn't say anything about its elements (required them packaged as a jar)
                    - Doesn't say anything about org.gradle.plugin.api-version (required '8.0')

    * Try:
    > Run with --stacktrace option to get the stack trace.
    > Run with --info or --debug option to get more log output.
    > Run with --scan to get full insights.

    * Get more help at https://help.gradle.org

    BUILD FAILED in 4s
    ➜ Samples git:(main) ✗ java --version
    openjdk 11.0.0.1 2023-05-09
    OpenJDK Runtime Environment 18.9 (build 11.0.0.1+3-5)
    OpenJDK 64-Bit Server VM 18.9 (build 11.0.0.1+3-5, mixed mode)

**Comment by ndr-brt** [comment, marked as answer]
java 11 is not supported anymore, please use java 17+ and eventually report the error that you got with that version
